dspace-server-webapp/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -572,6 +572,37 @@ public <ID extends Serializable> ResponseEntity<RepresentationModel<?>> upload(H
         return uploadInternal(request, apiCategory, model, uuid, uploadfile);
     }
 
+    /**
+     * Called in POST, multipart, upload to a specific rest resource the file passed as "file" request parameter
+     *
+     * Note that the regular expression in the request mapping accept a String as identifier;
+     *
+     * @param request
+     *            the http request
+     * @param apiCategory
+     *            the api category
+     * @param model
+     *            the rest model that identify the REST resource collection
+     * @param id
+     *            the id of the specific rest resource
+     * @param uploadfile
+     *            the file to upload
+     * @return the created resource
+     * @throws HttpRequestMethodNotSupportedException
+     */
+    @RequestMapping(method = RequestMethod.POST,
+            value = REGEX_REQUESTMAPPING_IDENTIFIER_AS_STRING_VERSION_STRONG,
+            headers = "content-type=multipart/form-data")
+    public <ID extends Serializable> ResponseEntity<RepresentationModel<?>> upload(HttpServletRequest request,
```

Comment:
If possible, we should add tests for this new method to `RestResourceControllerIT`

Reply:
methods `RestResourceController.delete(..., String);` | `RestResourceController.upload(..., String);` | `RestResourceController.patch(..., String);` have been added because they are used into `QAEventRestRepository`. Goal is the reflection is able to manage these calls and handle them properly to `QAEventRestRepository`. Consider **`QAEventRestRepositoryIT`** tests these methods already.